### PR TITLE
Do not raise exception in case of out-of-sequence Faucet event

### PR DESCRIPTION
**Pull request by didovesei**
Every time Forch connects to the socket it will restore the states from varzs. So I think we can remove the exception here.

### CODE REVIEW

**Review by grafnu — COMMENTED** [review]
Have the implications of this been sorted through with Ops & Eng?  Basically, with this PR the system will get into an inconsistent state when this happens... which really means that the process should be restarted or refreshed -- e.g. exit and full restart, or restore state from metrics.  Or, need explicit agreement from the team (eng/ops) that "it's ok that Forch will be inconsistent until somebody deals with the metric being incremented"

Basically, has the interstitial time from when this error occurs to something being done about it been thought about and clarified?

**Comment by didovesei** [comment]
> which really means that the process should be restarted or refreshed -- e.g. exit and full restart, or restore state from metrics

I agree that Eng/Ops should have been engaged. Just a clarification before involving them. I think we are already restoring the states from metrics. Do we still need to restart Forch or Faucet when there is an out-of-sequence event, in order for Forch to have states consitent with Faucet?

**Comment by grafnu** [comment]
Hmmm.... thinking about this more, maybe it wasn't necessary to engage them
previously if we have a clean solution in mind -- I was operating under the
assumption that things were *not* automatically restored when this happens.
But, I think this means two things:

   - We need some kind of test to make sure that when this condition
   happens that state is restored. Not 100% sure how to do this, but, it
   likely will require an artificial (not Faucet) event source that can be
   primed to feed events that trigger this, and then some way to check that
   the "state restored" bit of code happened. Have a think about how to do
   this and we can see what makes sense... it's not obvious to me what the
   right path here would be (how to test it without excessive work).
   - Really then we just need to let them know saying "this condition
   should be handled internally to Faucet/Forch now, but we created this new
   metric to monitor if it's happening" -- not sure if there's anything they
   would do on their end with that.

If the state isn't restored internally to forch, then we need to either
make it so, or go the other route.  If it is restored, then we don't need
to restart (since it's functionally equivalent to restoring).

**Comment by grafnu** [comment]
Might be that it's sufficient in a test to inject a single bad event (so
sequence number will be bad), and then make sure that the "restore state"
function is called... Seems like that should be a simple unit test -- if
not, we should likely figure out what to do to make it a simple unit test!

**Comment by didovesei** [comment]
> Might be that it's sufficient in a test to inject a single bad event (so sequence number will be bad), and then make sure that the "restore state" function is called... Seems like that should be a simple unit test -- if not, we should likely figure out what to do to make it a simple unit test!

Added unit test in "test_event.py" to test if "restore_states()" is called in case of out-of-sequence event.

File: testing/python_lib/test_event.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            self._forchestrator._faucet_events_connect()
+        except Exception as error:
+            print(f'Ignoring expected exception during restoring states: {error}')
+
+        self._forchestrator._faucet_events.set_event_horizon(100)
+        restore_states_called = False
+
+        try:
+            self._forchestrator.main_loop()
+        except InvalidConfigError as error:
```
Comment (grafnu) [inline testing/python_lib/test_event.py:69]: can you please add a comment explaining the logic here. I think it's that calling restore_states generates some other kind of error since the system is not properly configured?  Just a simple one line description of WHY this is would be great.  Otherwise, InvalidConfigError != FaucetEventOrderError so it seems off.

**Review by grafnu — COMMENTED** [review]
One minor comment to add a comment... otherwise, great way to test this!

File: testing/python_lib/test_event.py (hunk 1 above)
Comment (didovesei) [reply testing/python_lib/test_event.py:68]: Yes,  restore_states() will generate InvalidConfigError since Faucet config file is empty. Added the explanation in the comment..

**Review by grafnu — APPROVED** [review]
